pkg/epp/common/config/loader/configloader.go

```diff
@@ -64,18 +66,39 @@ func LoadSchedulerConfig(configProfiles []configapi.SchedulingProfile, handle pl
 	profiles := map[string]*framework.SchedulerProfile{}
 	for _, namedProfile := range configProfiles {
 		profile := framework.NewSchedulerProfile()
+		pickerAdded := false
+		scorerAdded := false
 		for _, plugin := range namedProfile.Plugins {
 			referencedPlugin := handle.Plugin(plugin.PluginRef)
 			if scorer, ok := referencedPlugin.(framework.Scorer); ok {
-				if plugin.Weight == nil {
-					return nil, fmt.Errorf("scorer '%s' is missing a weight", plugin.PluginRef)
+				// Set default weight to 1
```

Comment:
I strongly recommend consolidating all defaulting in the defaults.go for two reasons: 1) it is harder to reason about the default behavior and debug it when it is scattered 2) logging the actual configuration to be applied become easier and the log at https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/cmd/epp/runner/runner.go#L407 would show all the defaults we apply

Reply:
It is not about the weight only, I am referring to the whole logic we added here, including adding a default picker and default handler. I think all that should be done in the defaulting file.

Reply:
Done. I updated the code, tests, doc, and description of this PR